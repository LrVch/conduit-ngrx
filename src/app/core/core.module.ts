import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
