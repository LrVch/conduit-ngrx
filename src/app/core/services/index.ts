export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
