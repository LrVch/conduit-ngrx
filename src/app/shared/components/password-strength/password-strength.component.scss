.password-strength {
  display: flex;
  flex-direction: column-reverse;
  justify-content: start;
  position: absolute;
  right: -12px;
  bottom: 5px;
  width: 11px;

  div {
    width: 100%;
    height: 3px;
    margin-top: 3px;
    &:last-child {
      margin-top: 0;
    }
  }
}

.worst-pass {
  div {
    background-color: #f44336;
  }
}

.bad-pass {
  div {
    background-color: #f0d333;
  }
}

.weak-pass {
  div {
    background-color: #ff9800;
  }
}

.good-pass {
  div {
    background-color: #36bf3b;
  }
}

.strong-pass {
  div {
    background-color: #673ab7;
  }
}
