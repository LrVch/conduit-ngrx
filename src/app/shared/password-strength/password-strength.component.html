<div class="password-strength" [ngClass]="clases">
  <div *ngFor="let item of items"></div>
</div>
