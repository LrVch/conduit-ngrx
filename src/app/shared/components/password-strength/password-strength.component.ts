import { Component, Input } from '@angular/core';
// import * as zxcvbn from 'zxcvbn';

// const zxcvbnFn = zxcvbn || (<any>window).zxcvbn;

enum strength {
  worst = 1,
  bad = 2,
  weak = 3,
  good = 4,
  strong = 5
}

@Component({
  selector: 'app-password-strength',
  templateUrl: './password-strength.component.html',
  styleUrls: ['./password-strength.component.scss']
})
export class PasswordStrengthComponent {
  @Input() set password(password: string) {
    if (!password) {
      this.items = [];
      return;
    }
    // this.result = zxcvbnFn(password).score + 1;
    // this.items = Array.from({ length: this.result });
  }

  result: number;
  items: any[];

  get clases() {
    return {
      'worst-pass': this.result === strength.worst,
      'bad-pass': this.result === strength.bad,
      'weak-pass': this.result === strength.weak,
      'good-pass': this.result === strength.good,
      'strong-pass': this.result === strength.strong
    };
  }
}
