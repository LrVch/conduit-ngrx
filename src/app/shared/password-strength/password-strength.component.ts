import { Component, Input } from '@angular/core';
import zxcvbn from 'zxcvbn';

@Component({
  selector: 'app-password-strength',
  templateUrl: './password-strength.component.html',
  styleUrls: ['./password-strength.component.scss']
})
export class PasswordStrengthComponent {
  @Input() set password(password: string) {
    if (!password) {
      this.items = [];
      return;
    }
    this.result = zxcvbn(password).score + 1;
    this.items = Array.from({length: this.result});
  }

  result: number;
  items: any[];

  get clases() {
    return {
      'worst-pass': this.result === 1,
      'bad-pass': this.result === 2,
      'weak-pass': this.result === 3,
      'good-pass': this.result === 4,
      'strong-pass': this.result === 5,
    };
  }
}
