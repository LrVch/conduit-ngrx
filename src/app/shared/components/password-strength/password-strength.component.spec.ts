import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '../../lib/testing';

import { PasswordStrengthComponent } from './password-strength.component';
import { NO_ERRORS_SCHEMA, DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';

describe('PasswordStrengthComponent', () => {
    let component: PasswordStrengthComponent;
    let fixture: ComponentFixture<PasswordStrengthComponent>;
    let de: DebugElement;
    let el: HTMLElement;

    const strength = {
        worst: '1234',
        bad: 'qswd',
        weak: 'qswdefr',
        good: 'qswdefrgt',
        strong: 'qswdefrgt_1'
    };

    beforeEach(
        async(() => {
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [PasswordStrengthComponent]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(PasswordStrengthComponent);
                component = fixture.componentInstance;
                de = fixture.debugElement;
                el = de.nativeElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should set items to empty array and result to be undefined by "password" @Input', () => {
        component.password = '';
        fixture.detectChanges();

        expect(component.items).toEqual([]);
        expect(component.result).toBe(undefined);
    });

    it('should set 1 item to item\'s array and result to be "1" by "password" @Input', () => {
        component.password = strength.worst;
        fixture.detectChanges();

        expect(component.result).toBe(1);
        expect(component.items).toEqual(Array.from({ length: 1 }));
    });

    it('should set 2 item to item\'s array and result to be "2" by "password" @Input', () => {
        component.password = strength.bad;
        fixture.detectChanges();

        expect(component.result).toBe(2);
        expect(component.items).toEqual(Array.from({ length: 2 }));
    });

    it('should set 3 item to item\'s array and result to be "3" by "password" @Input', () => {
        component.password = strength.weak;
        fixture.detectChanges();

        expect(component.result).toBe(3);
        expect(component.items).toEqual(Array.from({ length: 3 }));
    });

    it('should set 4 item to item\'s array and result to be "4" by "password" @Input', () => {
        component.password = strength.good;
        fixture.detectChanges();

        expect(component.result).toBe(4);
        expect(component.items).toEqual(Array.from({ length: 4 }));
    });
    it('should set 5 item to item\'s array and result to be "5" by "password" @Input', () => {
        component.password = strength.strong;
        fixture.detectChanges();

        expect(component.result).toBe(5);
        expect(component.items).toEqual(Array.from({ length: 5 }));
    });

    it('shouldn\'t show any indicators', () => {
        component.password = '';
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should show one indicators', () => {
        component.password = strength.worst;
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should show two indicators', () => {
        component.password = strength.bad;
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should show three indicators', () => {
        component.password = strength.weak;
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should show four indicators', () => {
        component.password = strength.good;
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should show five indicators', () => {
        component.password = strength.strong;
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });
});

