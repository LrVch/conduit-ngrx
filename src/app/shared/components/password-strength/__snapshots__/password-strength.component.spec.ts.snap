// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordStrengthComponent should show five indicators 1`] = `
<app-password-strength
  items={[Function Array]}
  result={[Function Number]}
>
  <div
    class="password-strength strong-pass"
    ng-reflect-klass="password-strength"
    ng-reflect-ng-class="[object Object]"
  >
    
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</app-password-strength>
`;

exports[`PasswordStrengthComponent should show four indicators 1`] = `
<app-password-strength
  items={[Function Array]}
  result={[Function Number]}
>
  <div
    class="password-strength good-pass"
    ng-reflect-klass="password-strength"
    ng-reflect-ng-class="[object Object]"
  >
    
    <div />
    <div />
    <div />
    <div />
  </div>
</app-password-strength>
`;

exports[`PasswordStrengthComponent should show one indicators 1`] = `
<app-password-strength
  items={[Function Array]}
  result={[Function Number]}
>
  <div
    class="password-strength worst-pass"
    ng-reflect-klass="password-strength"
    ng-reflect-ng-class="[object Object]"
  >
    
    <div />
  </div>
</app-password-strength>
`;

exports[`PasswordStrengthComponent should show three indicators 1`] = `
<app-password-strength
  items={[Function Array]}
  result={[Function Number]}
>
  <div
    class="password-strength weak-pass"
    ng-reflect-klass="password-strength"
    ng-reflect-ng-class="[object Object]"
  >
    
    <div />
    <div />
    <div />
  </div>
</app-password-strength>
`;

exports[`PasswordStrengthComponent should show two indicators 1`] = `
<app-password-strength
  items={[Function Array]}
  result={[Function Number]}
>
  <div
    class="password-strength bad-pass"
    ng-reflect-klass="password-strength"
    ng-reflect-ng-class="[object Object]"
  >
    
    <div />
    <div />
  </div>
</app-password-strength>
`;

exports[`PasswordStrengthComponent shouldn't show any indicators 1`] = `
<app-password-strength
  items={[Function Array]}
>
  <div
    class="password-strength"
    ng-reflect-klass="password-strength"
    ng-reflect-ng-class="[object Object]"
  >
    
  </div>
</app-password-strength>
`;
